src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/SpatialCoordinateSystemParticipant.cs

```diff
@@ -193,5 +205,33 @@ internal void ReadCoordinateStateMessage(BinaryReader reader)
             PeerSpatialCoordinateWorldPosition = reader.ReadVector3();
             PeerSpatialCoordinateWorldRotation = reader.ReadQuaternion();
         }
+
+        internal void SendSupportedLocalizersMessage(SocketEndpoint endpoint, ICollection<Guid> supportedLocalizers)
+        {
+            using (MemoryStream stream = new MemoryStream())
+            using (BinaryWriter writer = new BinaryWriter(stream))
+            {
+                writer.Write(SpatialCoordinateSystemManager.SupportedLocalizersMessageHeader);
+                writer.Write(supportedLocalizers.Count);
+                foreach (Guid supportedLocalizer in supportedLocalizers)
+                {
+                    writer.Write(supportedLocalizer);
+                }
+
+                endpoint.Send(stream.ToArray());
+            }
+        }
+
+        internal void ReadSupportedLocalizersMessage(BinaryReader reader)
+        {
+            int localizerCount = reader.ReadInt32();
+            HashSet<Guid> supportedLocalizers = new HashSet<Guid>();
+            for (int i = 0; i < localizerCount; i++)
+            {
+                supportedLocalizers.Add(reader.ReadGuid());
+            }
+
+            peerSupportedLocalizersTaskSource.SetResult(supportedLocalizers);
```

Comment:
TrySetResult, as the dispose might cancel it and this may still get a callback as I understand.